core/src/main/java/com/novoda/noplayer/player/PlayerFactory.java

```diff
@@ -18,78 +19,75 @@
-                return createExoPlayer(createDrmSessionCreatorFor(drm));
+                DrmSessionCreator drmSessionCreator = createDrmSessionCreatorFor(drmType, drmHandler);
+                return exoPlayerCreator.createExoPlayer(context, drmSessionCreator);
             default:
                 throw UnableToCreatePlayerException.unknownPlayerType(playerType);
         }
     }
 
-    private DrmSessionCreator createDrmSessionCreatorFor(DrmHandler drm) {
-        if (drm == DrmHandler.NO_DRM) {
-            return new NoDrmSessionCreator();
-        } else if (drm instanceof StreamingModularDrm) {
-            ProvisionExecutor provisionExecutor = ProvisionExecutor.newInstance();
-            ProvisioningModularDrmCallback mediaDrmCallback = new ProvisioningModularDrmCallback((StreamingModularDrm) drm, provisionExecutor);
-            return new StreamingDrmSessionCreator(mediaDrmCallback);
-        } else if (drm instanceof DownloadedModularDrm) {
-            return new LocalDrmSessionCreator((DownloadedModularDrm) drm);
-        } else {
-            throw UnableToCreatePlayerException.unknownDrmHandler(drm);
+    private DrmSessionCreator createDrmSessionCreatorFor(DrmType drmType, DrmHandler drmHandler) {
+        switch (drmType) {
+            case NONE:
+            case WIDEVINE_CLASSIC:
+                return new NoDrmSessionCreator();
+            case WIDEVINE_MODULAR_STREAM:
+                ProvisionExecutor provisionExecutor = ProvisionExecutor.newInstance();
+                ProvisioningModularDrmCallback mediaDrmCallback = new ProvisioningModularDrmCallback((StreamingModularDrm) drmHandler, provisionExecutor);
+                return new StreamingDrmSessionCreator(mediaDrmCallback);
+            case WIDEVINE_MODULAR_DOWNLOAD:
+                return new LocalDrmSessionCreator((DownloadedModularDrm) drmHandler);
+            default:
+                throw UnableToCreatePlayerException.noDrmHandlerFor(drmType);
         }
     }
 
-    private Player createExoPlayer(DrmSessionCreator drmSessionCreator) {
-        RendererFactory rendererFactory = createRendererFactory(drmSessionCreator);
-        ExoPlayerFacade exoPlayerFacade = new ExoPlayerFacade(rendererFactory);
-        return new ExoPlayerImpl(exoPlayerFacade);
-    }
-
-    private RendererFactory createRendererFactory(DrmSessionCreator drmSessionCreator) {
-        return new RendererFactory(context, drmSessionCreator);
-    }
-
-    private Player createMediaPlayer() {
-        AndroidMediaPlayerFacade androidMediaPlayer = new AndroidMediaPlayerFacade(context);
-        return new AndroidMediaPlayerImpl(androidMediaPlayer);
-    }
-
     static class UnableToCreatePlayerException extends RuntimeException {
 
-        static UnableToCreatePlayerException contentNotSupported() {
-            return new UnableToCreatePlayerException("No player available to handle content");
+        static UnableToCreatePlayerException unknownDrmType(DrmType drmType) {
```

Comment:
commit f992442dde8f0694861d8efda365b6790ad908ce
Author: Ataul Munim <hello@ataulm.com>
Date:   Mon Jun 5 11:43:15 2017 +0100

    Rename exception static factories to match error messages